### PR TITLE
Enchantments, vines, particles 11a

### LINKED ISSUE #1084: Incorrect enchantment method names
- `EnchantmentHelper.contains(Collection<Enchantment>, Enchantment)` -> `hasNoConflicts`
  - The method returns true if the `Collection<Enchantment>` does not have conflicting enchantments.
- `Enchantment.isDifferent(Enchantment)` -> "can be on the same item"
  - basically `this.differs(other) && other.differs(this)`. `differs` itself is named wrong, so this method checks that both enchantments think the other one is a valid partner.
  - Example: `SilkTouchEnchantment.differs` returns false if the enchantment is `FORTUNE`, so they conflict even though Fortune uses the default impl.
- `Enchantment.differs(Enchantment)` -> also "can be on the same item"
  - Checks if this enchantment is mutually exclusive with the passed enchantment.

### PR DESCRIPTION
**Pull request by liach**
Fixes #1084 as well

@SuperCoder7979 that level generator is left for you

### CODE REVIEW

**Review by YanisBft — APPROVED** [review]

**Review by Runemoro — APPROVED** [review]

File: mappings/net/minecraft/world/gen/feature/TwistingVinesFeature.mapping [hunk 1]
```diff
@@ -0,0 +1,15 @@
+CLASS net/minecraft/class_4953 net/minecraft/world/gen/feature/TwistingVinesFeature
+	METHOD method_25986 isNotSuitable (Lnet/minecraft/class_1936;Lnet/minecraft/class_2338;)Z
+		ARG 1 world
+		ARG 2 pos
+	METHOD method_25987 generateColumn (Lnet/minecraft/class_1936;Ljava/util/Random;Lnet/minecraft/class_2338$class_2339;III)V
```
Comment (jaskarth) [inline mappings/net/minecraft/world/gen/feature/TwistingVinesFeature.mapping]:
```suggestion
	METHOD method_25987 generateVines (Lnet/minecraft/class_1936;Ljava/util/Random;Lnet/minecraft/class_2338$class_2339;III)V
```

File: mappings/net/minecraft/world/gen/feature/TwistingVinesFeature.mapping [hunk 2]
```diff
@@ -0,0 +1,15 @@
+CLASS net/minecraft/class_4953 net/minecraft/world/gen/feature/TwistingVinesFeature
+	METHOD method_25986 isNotSuitable (Lnet/minecraft/class_1936;Lnet/minecraft/class_2338;)Z
```
Comment (SuperCoder7979) [inline mappings/net/minecraft/world/gen/feature/TwistingVinesFeature.mapping]:
```suggestion
	METHOD method_25986 canGenerate (Lnet/minecraft/class_1936;Lnet/minecraft/class_2338;)Z
```

File: mappings/net/minecraft/world/gen/feature/TwistingVinesFeature.mapping [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+		ARG 1 world
+		ARG 2 pos
+	METHOD method_25987 generateColumn (Lnet/minecraft/class_1936;Ljava/util/Random;Lnet/minecraft/class_2338$class_2339;III)V
+		ARG 0 world
+		ARG 1 random
+		ARG 2 pos
+		ARG 3 maxLength
+		ARG 4 minAge
+		ARG 5 maxAge
+	METHOD method_25988 generatePile (Lnet/minecraft/class_1936;Ljava/util/Random;Lnet/minecraft/class_2338;)V
```
Comment (jaskarth) [inline mappings/net/minecraft/world/gen/feature/TwistingVinesFeature.mapping]:
```suggestion
	METHOD method_25988 tryGenerateVines (Lnet/minecraft/class_1936;Ljava/util/Random;Lnet/minecraft/class_2338;)V
```

**Review by jaskarth — CHANGES_REQUESTED** [review]

File: mappings/net/minecraft/world/gen/feature/TwistingVinesFeature.mapping (hunk 3 above)
Comment (liach) [reply mappings/net/minecraft/world/gen/feature/TwistingVinesFeature.mapping]: No, it generates multiple columns of vines in this method.

**Comment by liach** [comment]
@SuperCoder7979 seriously you need to read the code

File: mappings/net/minecraft/world/gen/feature/TwistingVinesFeature.mapping (hunk 1 above)
Comment (liach) [reply mappings/net/minecraft/world/gen/feature/TwistingVinesFeature.mapping]: This is generating a single column of vine. `generateVines` make people unsure if it is generating a group of columns or a single column.

File: mappings/net/minecraft/world/gen/feature/TwistingVinesFeature.mapping (hunk 3 above)
Comment (liach) [inline mappings/net/minecraft/world/gen/feature/TwistingVinesFeature.mapping]:
```suggestion
	METHOD method_25988 generateVinesInArea (Lnet/minecraft/class_1936;Ljava/util/Random;Lnet/minecraft/class_2338;)V
```

File: mappings/net/minecraft/world/gen/feature/TwistingVinesFeature.mapping (hunk 1 above)
Comment (liach) [reply mappings/net/minecraft/world/gen/feature/TwistingVinesFeature.mapping]:
```suggestion
	METHOD method_25987 generateVineColumn (Lnet/minecraft/class_1936;Ljava/util/Random;Lnet/minecraft/class_2338$class_2339;III)V
```

File: mappings/net/minecraft/block/Vines.mapping [hunk 4]
```diff
@@ -0,0 +1,4 @@
+CLASS net/minecraft/class_4949 net/minecraft/block/Vines
```
Comment (jaskarth) [inline mappings/net/minecraft/block/Vines.mapping]: `VineHelper` maybe? just `Vines` seems too ambiguous to me

Comment (liach) [reply mappings/net/minecraft/block/Vines.mapping]: i use this name as we have Texts class. this holds a collision shape and an air check for root for those who want to suggest alternative names.

**Review by jaskarth — APPROVED** [review]
looks good!

**Review by Runemoro — APPROVED** [review]

**Review by Prospector — APPROVED** [review]

**Pull request merged**